File: oms/src/main/webapp/resources/js/service/customer_service.js
'use strict';

omsApp.factory('CustomerService', ['$http', '$q','CommonService', function($http, $q, CommonService){

	return {
		
			fetchAllCustomers: function() {
					return $http.get(CommonService.getBaseUrl()+'/service/customer/')
							.then(
									function(response){
										return response.data;
									}, 
									function(errResponse){
										console.error('Error while fetching customers');
										return $q.reject(errResponse);
									}
							);
			},
			
			addCustomer: function(customer) {
				return $http.post(CommonService.getBaseUrl()+'/service/customer/', customer)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while adding customer');
							return $q.reject(errResponse);
						}
				);

			},
			updateCustomer: function(customer) {
				return $http.put(CommonService.getBaseUrl()+'/service/customer/', customer)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while updating customer');
							return $q.reject(errResponse);
						}
				);

			},

			deleteCustomer: function(customerId) {
				return $http.delete(CommonService.getBaseUrl()+'/service/customer/'+customerId)
				.then(
						function(response){
							return response.data;
						}, 
						function(errResponse){
							console.error('Error while fetching customer count');
							return $q.reject(errResponse);
						}
				);

			},

			getCustomerCount: function() {
			return $http.get(CommonService.getBaseUrl()+'/service/customer/count')
					.then(
							function(response){
								return response.data;
							}, 
							function(errResponse){
								console.error('Error while fetching customer count');
								return $q.reject(errResponse);
							}
					);
		    
			}
		  
	};

}]);

File: oms/src/main/java/com/vjentrps/oms/service/CustomerService.java
package com.vjentrps.oms.service;

import java.util.List;

import com.vjentrps.oms.exception.OmsServiceException;
import com.vjentrps.oms.model.BasicInfo;
import com.vjentrps.oms.model.Customer;

public interface CustomerService {
	
	 void addCustomer(Customer customer) throws OmsServiceException;
	
	 void deleteCustomer(long customerId) throws OmsServiceException;
	
	 void updateCustomer(Customer customer) throws OmsServiceException;
	 
	 List<Customer> listCustomers() throws OmsServiceException;

	Customer getCustomerById(long customerId) throws OmsServiceException;

	int getCustomerCount() throws OmsServiceException;

	List<BasicInfo> getCustomersBasicInfo() throws OmsServiceException;

	 
	
}
